### PR TITLE
feat: add emoji reaction on comments

### PR DESCRIPTION
**Pull request by kopkop123**

### CODE REVIEW

File: backend/src/common/interfaces/emoji/emoji.interface.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+interface IEmoji {
```
Comment (ghost) [inline backend/src/common/interfaces/emoji/emoji.interface.ts]: Why do we need this file if we export IEmoji from shared?

File: backend/src/common/interfaces/comment-reaction/comment-reaction.interface.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+interface ICommentReaction {
```
Comment (ghost) [inline backend/src/common/interfaces/comment-reaction/comment-reaction.interface.ts]: Why do we need this file if we export ICommentReaction from shared?

File: backend/src/data/repositories/comment-reaction.repository.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { EntityRepository, Repository, In } from 'typeorm';
+import { Reaction } from '../entities/reaction';
+
+@EntityRepository(Reaction)
+class CommentReactionRepository extends Repository<Reaction> {
+  public getAllReactionsByCommentId(commentId: string): Promise<Reaction[]> {
```
Comment (ghost) [inline backend/src/data/repositories/comment-reaction.repository.ts]: No need of creation this function, use this.find({ commentId }) directly

File: backend/src/data/repositories/comment-reaction.repository.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    return this.find({ commentId });
+  }
+
+  public getReactionsById(ids: string[]): Promise<Reaction[]> {
+    return this.find({ where: { id: In(ids) } });
+  }
+
+  public findById(id: string): Promise<Reaction> {
```
Comment (ghost) [inline backend/src/data/repositories/comment-reaction.repository.ts]: No need of creation this function, use this.find({ id }) directly

File: frontend/src/components/pages/components/emoji/emoji.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
+import { MutableRefObject, SyntheticEvent } from 'react';
+import { Dropdown, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { useAppSelector, useEffect, useState, useRef } from 'hooks/hooks';
+import { CommentReactionApi } from 'services';
+import Picker from 'emoji-picker-react';
+import { ICommentReaction } from '../../../../common/interfaces/comment-reaction';
+import { IEmoji } from '../../../../common/interfaces/emoji';
```
Comment (ghost) [inline frontend/src/components/pages/components/emoji/emoji.tsx]: Use aliases

File: frontend/src/components/pages/components/emoji/emoji.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
+import { MutableRefObject, SyntheticEvent } from 'react';
+import { Dropdown, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { useAppSelector, useEffect, useState, useRef } from 'hooks/hooks';
+import { CommentReactionApi } from 'services';
+import Picker from 'emoji-picker-react';
```
Comment (ghost) [inline frontend/src/components/pages/components/emoji/emoji.tsx]: Must be on top of imports

**Review by ghost — APPROVED** [review]

File: backend/src/api/routes/index.ts [hunk 7]
```diff
@@ -6,6 +6,7 @@ import workspaceRoute from './workspace-route';
 import tagRoute from './tag-route';
 import teamRoute from './team-route';
 import skillRoute from './skill-route';
+import commentReactionRoute from './comment-reaction-route';
```
Comment (AlikRakhmonov) [inline backend/src/api/routes/index.ts]:
```suggestion
import commentReactionRoute from './comment-route';
```

File: backend/src/api/routes/comment-reaction-route.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import { Router } from 'express';
+import { run } from '../../common/helpers/route.helper';
+import {
+  getAllCommentReactions,
+  handleCommentReaction,
+} from '../../services/comment-reaction.service';
```
Comment (AlikRakhmonov) [inline backend/src/api/routes/comment-reaction-route.ts]:
```suggestion
} from '../../services/comment.service';
```

**Review by AlikRakhmonov — DISMISSED** [review]

File: backend/src/common/interfaces/emoji/emoji.interface.ts (hunk 1 above)
Comment (kopkop123) [reply backend/src/common/interfaces/emoji/emoji.interface.ts]: fixed

File: backend/src/common/interfaces/comment-reaction/comment-reaction.interface.ts (hunk 2 above)
Comment (kopkop123) [reply backend/src/common/interfaces/comment-reaction/comment-reaction.interface.ts]: fixed

File: backend/src/api/routes/index.ts (hunk 7 above)
Comment (kopkop123) [reply backend/src/api/routes/index.ts]: fixed

File: backend/src/api/routes/comment-reaction-route.ts (hunk 8 above)
Comment (kopkop123) [reply backend/src/api/routes/comment-reaction-route.ts]: fixed

File: frontend/src/components/pages/components/comment/comment.tsx [hunk 9]
```diff
@@ -3,8 +3,10 @@ import { useState, useHistory } from 'hooks/hooks';
 import { replaceIdParam } from 'helpers/helpers';
 import { AppRoute } from 'common/enums/enums';
 import { CommentForm } from '../components';
+import { Emoji } from '../emoji/emoji';
 import { UserAvatar } from 'components/common/common';
 import { getAllowedClasses } from 'helpers/helpers';
+import { ICommentReaction } from '../../../../common/interfaces/comment-reaction';
```
Comment (ursarik) [inline frontend/src/components/pages/components/comment/comment.tsx]:
```suggestion
import { ICommentReaction } from 'common/interfaces/comment-reaction';
```

**Review by ovchynnikov — APPROVED** [review]

**Review by ursarik — APPROVED** [review]

File: frontend/src/components/pages/components/emoji/emoji.tsx (hunk 6 above)
Comment (kopkop123) [reply frontend/src/components/pages/components/emoji/emoji.tsx]: fixed

File: frontend/src/components/pages/components/emoji/emoji.tsx (hunk 5 above)
Comment (kopkop123) [reply frontend/src/components/pages/components/emoji/emoji.tsx]: fixed

File: backend/src/data/repositories/comment-reaction.repository.ts (hunk 4 above)
Comment (kopkop123) [reply backend/src/data/repositories/comment-reaction.repository.ts]: fixed

File: backend/src/data/repositories/comment-reaction.repository.ts (hunk 3 above)
Comment (kopkop123) [reply backend/src/data/repositories/comment-reaction.repository.ts]: fixed

**Pull request merged**